src/ui/public/config/defaults.js

```diff
@@ -59,6 +59,10 @@ define(function (require) {
         value: 500,
         description: 'The number of rows to show in the table',
       },
+      'discover:highlight': {
```

Comment:
Given that this affects both discover and dashboard, this should be `doc_table:highlight`